File: 2020/13/main.R
# this doesn't work for large numbers :(
"input <-
  scan(\"c:\\Work\\Code\\AdventOfCode\\2020\\13\\test.txt\",
       character(),
       sep = \",\")
numbers <- sort(strtoi(input[input != \"x\"]), decreasing = TRUE)
offset <- match(numbers[1], input) - 1
timetable <- seqMpfr(9.999e+14, 1e+15, by = numbers[1]) - offset
for (i in 2:length(numbers)) {
  offset <- match(numbers[i], input) - 1
  t <-
    seqMpfr(timetable[1], 100000000000000, by = numbers[i]) - offset
  timetable <- intersect(t, timetable)
}
print(timetable[1])
"
input <-
  scan("c:\\Work\\Code\\AdventOfCode\\2020\\13\\test.txt",
       character(),
       sep = ",")
numbers <- sort(strtoi(input[input != "x"]), decreasing = TRUE)
offset <- match(numbers[1], input) - 1
stepper <- numbers[1]
cursor <- 0
i<- 2
while (TRUE) {
  cursor <- cursor + offset + stepper
  check_against <- (match(numbers[i], input) - 1)
  meh <- cursor - offset + check_against
  if (meh %% numbers[i] == 0)
  {
    stepper <- meh - check_against
    i<-i+1
    if (i > length(numbers)) {
      break
    }
  }
}
check <- offset + stepper

File: 2020/13/main.dart
import 'dart:io';
import '../base/handy_utils.dart';

void main() {
  //var times = ["7", "13", "x", "x", "59", "x", "31", "19"];
  var inputFile = new File("c:\\Work\\Code\\AdventOfCode\\2020\\13\\input.txt");
  var times = inputFile.readAsStringSync().trim().split(",");
  int t = 0;
  int increment = int.parse(times[0]);
  int offset = 1;
  while (offset < times.length) {
    if (times[offset] == "x") {
      offset++;
      continue;
    }
    t += increment;
    var current = int.parse(times[offset]);
    if ((t + offset) % current != 0) {
      continue;
    }
    increment *= current;
    offset++;
  }
  print(times);
  print(t);
}
